src/sync/impl/apple/system_configuration.hpp

```diff
@@ -0,0 +1,70 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_SYSTEM_CONFIGURATION_HPP
+#define REALM_OS_SYSTEM_CONFIGURATION_HPP
+
+#include "sync/impl/network_reachability.hpp"
+
+#if NETWORK_REACHABILITY_AVAILABLE
+
+#include <SystemConfiguration/SystemConfiguration.h>
+
+namespace realm {
+
+class SystemConfiguration {
```

Comment:
Things in the `impl` directory are typically contained within the `realm::_impl` namespace rather than directly in `realm`.